mobile_config/src/admin_service.rs

```diff
@@ -0,0 +1,159 @@
+use crate::{
+    key_cache::{self, CacheKeys, KeyCache, KeyType},
+    settings::Settings,
+    GrpcResult,
+};
+use anyhow::{anyhow, Result};
+use chrono::Utc;
+use file_store::traits::{MsgVerify, TimestampEncode};
+use futures::future::TryFutureExt;
+use helium_crypto::{Keypair, Network, PublicKey, PublicKeyBinary, Sign};
+use helium_proto::{
+    services::mobile_config::{self, AdminAddKeyReqV1, AdminKeyResV1, AdminRemoveKeyReqV1},
+    Message,
+};
+use sqlx::{Pool, Postgres};
+use tokio::sync::watch;
+use tonic::{Request, Response, Status};
+
+pub struct AdminService {
+    key_cache: KeyCache,
+    key_cache_updater: watch::Sender<CacheKeys>,
+    pool: Pool<Postgres>,
+    required_network: Network,
```

Comment:
nit:  "required" seems redundant here? Just use network and either crash or don't do anything if the network is not present in settings?

Also, naming wise, should this be "network" or "cluster"?